grpclb/src/main/java/io/grpc/grpclb/GrpclbLoadBalancer.java

```diff
@@ -74,21 +75,26 @@
   public void handleResolvedAddresses(ResolvedAddresses resolvedAddresses) {
     Attributes attributes = resolvedAddresses.getAttributes();
     List<EquivalentAddressGroup> newLbAddresses = attributes.get(GrpclbConstants.ATTR_LB_ADDRS);
-    List<EquivalentAddressGroup> overrideAuthorityLbAddresses = newLbAddresses == null
-        ? Collections.<EquivalentAddressGroup>emptyList() :
-        Collections.unmodifiableList(newLbAddresses);
-    if (overrideAuthorityLbAddresses.isEmpty() && resolvedAddresses.getAddresses().isEmpty()) {
+    if (newLbAddresses == null) {
+      newLbAddresses = Collections.emptyList();
+    }
+    if (newLbAddresses.isEmpty() && resolvedAddresses.getAddresses().isEmpty()) {
       handleNameResolutionError(
           Status.UNAVAILABLE.withDescription("No backend or balancer addresses found"));
       return;
     }
-
-    for (EquivalentAddressGroup lbAddr : overrideAuthorityLbAddresses) {
+    List<EquivalentAddressGroup> overrideAuthorityLbAddresses =
+        new ArrayList<>(newLbAddresses.size());
+    for (EquivalentAddressGroup lbAddr : newLbAddresses) {
       String lbAddrAuthority = lbAddr.getAttributes().get(GrpclbConstants.ATTR_LB_ADDR_AUTHORITY);
       if (lbAddrAuthority == null) {
         throw new AssertionError(
             "This is a bug: LB address " + lbAddr + " does not have an authority.");
       }
+      Attributes attrs = Attributes.newBuilder()
```

Comment:
We should probably use `lbAttr.getAttributes().toBuilder()` to preserve the attributes we don't know about. We also need to preserve ATTR_LB_ADDR_AUTHORITY in the attributes because it is used by other parts of the system (GoogleDefaultCredentials) to detect gRPC-LB.